contracts/BackedOracleController.sol

```diff
@@ -0,0 +1,74 @@
+/**
+ * SPDX-License-Identifier: MIT
+ *
+ * Copyright (c) 2021-2022 Backed Finance AG
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+pragma solidity 0.8.9;
+
+import "./BackedOracle.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+
+contract BackedOracleController is AccessControl {
+  BackedOracle private _oracle;
+
+  bytes32 public constant UPDATER_ROLE = keccak256("UPDATER_ROLE");
+
+  int8 public constant MAX_PERCENT_DIFFERENCE = 10;
+  uint32 public constant MIN_UPDATE_INTERVAL = 1 hours;
+
+  constructor(BackedOracle oracle, address admin) {
+    _oracle = oracle;
+    _grantRole(DEFAULT_ADMIN_ROLE, admin);
+  }
+
+  function updateAnswer(int192 newAnswer, uint32 newTimestamp, uint32 newRound) public onlyRole(UPDATER_ROLE) {
+    (,int256 latestAnswer,, uint256 latestTimestamp,) = _oracle.latestRoundData();
```

Comment:
Since this would revert if there is no round set yet, the oracle's updateAnswer must be called first before making the Oracle Controller its owner, otherwise this will always revert. This seems a bit complicated, as you'd need to create the oracle, do initial value setting, then transfer the ownership, all in different calls, instead of a simple init process. This could be solved by doing a try call, and ignoring some of the constraints here when the answer is updated for the first time, or by uniting the contracts (see: https://github.com/backed-fi/backed-token-contract/pull/31/files#r1176795993)

Reply:
united the contracts dc683b0ec7df2a91033d121e24e0142d8da2a476